### PR TITLE
Meta tweaks

### PR DESCRIPTION
**Pull request by jameswomack**
* Block scoped variables / using `const` to communicate something isn't reassigned to
* Export using an arrow function
* Enhance error message
* Changed Travis config & NPM engines field accordingly

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
 'use strict';
 module.exports = function (str, opts) {
-	if (typeof str !== 'string') {
-		throw new TypeError('Expected a string in the first argument, got ' + typeof str);
+	const typeOfInitialArg = typeof str;
+	if (typeOfInitialArg !== 'string') {
```
Comment (SamVerschueren) [inline index.js]: Just use `typeof` like before (twice).

File: index.js [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+		throw new TypeError('Expected a string in the first argument, got ' + typeOfInitialArg);
```
Comment (SamVerschueren) [inline index.js]: Use template string

```js
throw new TypeError(`Expected input to be of type \`string\`, got \`${typeof str}\``);
```

File: index.js [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
 	}
 
-	var preserveNewlines = opts && opts.preserveNewlines;
+	const lineMatcher = /\r?\n/;
```
Comment (SamVerschueren) [inline index.js]: Don't really see any added value in extracting this. Line 14 is really ugly now. So just revert it.

File: index.js [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
 
-	if (!preserveNewlines) {
-		return str.split(/\r?\n/);
+	if (!(opts && opts.preserveNewlines)) {
```
Comment (SamVerschueren) [inline index.js]:
```js
opts = Object.assign({
    preserveNewLines: false
}, opts);

if (!opts.preserveNewLines) {

}
```

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Also update `.travis.yml` and the engines field in `package.json`.

File: index.js [hunk 5]
```diff
@@ -1,19 +1,20 @@
 'use strict';
 module.exports = function (str, opts) {
```
Comment (sindresorhus) [inline index.js]: Use arrow function here.

**Pull request merged**